runtime/testy/Ansible.py

```diff
@@ -0,0 +1,176 @@
+                    printStackTrace = True))
+
+    def udpSender(self, badThingsQueue, stateQueue, pipe):
+        """Function run as a thread that sends a packaged state from the TwoBuffer
+        
+        The current state that has already been packaged is gotten from the 
+        TwoBuffer, and is sent to Dawn via a UDP socket.
+        """
+        host = socket.gethostname() #TODO: determine host in runtime-dawn comm
+        with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
+            while True: 
+                try:
+                    msg = self.sendBuffer.get()
+                    if msg != 0: 
+                        s.sendto(msg, (host, UDPSendClass.SEND_PORT))
+                except Exception:
+                    badThingsQueue.put(BadThing(sys.exc_info(), 
+                    "UDP sender thread has crashed with error:",  
+                    event = BAD_EVENTS.UDP_SEND_ERROR, 
+                    printStackTrace = True))
+
+
+class UDPRecvClass(AnsibleHandler):
+    RECV_PORT = 1236
+    
+    def __init__(self, badThingsQueue, stateQueue, pipe):
+        self.recvBuffer = TwoBuffer()        
+        packName = THREAD_NAMES.UDP_UNPACKAGER
+        sockRecvName = THREAD_NAMES.UDP_RECEIVER
+        super().__init__(packName, sockRecvName, UDPRecvClass.unpackageData,
+                         UDPRecvClass.udpReceiver, badThingsQueue, stateQueue, pipe)
+
+    def udpReceiver(self, badThingsQueue, stateQueue, pipe):
+        """Function to receive data from Dawn to local TwoBuffer
+
+        Listens on the receive port and stores data into TwoBuffer to be shared
+        with the unpackager.
+        """
+        host = socket.gethostname() #TODO: determine host between dawn-runtime comm
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM) 
+        s.bind((host, UDPRecvClass.RECV_PORT))
+        while True:
+            try:
+                recv_data = s.recv(2048)
+                self.recvBuffer.replace(recv_data)
+            except Exception as e:
+                    badThingsQueue.put(BadThing(sys.exc_info(), 
+                    "UDP receiver thread has crashed with error:",  
+                    event = BAD_EVENTS.UDP_RECV_ERROR, 
+                    printStackTrace = True))
+
+    def unpackageData(self, badThingsQueue, stateQueue, pipe):
+        """Unpackages data from proto and sends to stateManager on the SM stateQueue
+
+        Sending data from dawn to stateManager is supported, the unpackage function is
+        currently unimplemented.  
+        """
+        def unpackage(data):
+            """Function that takes a packaged proto and unpackages item
+
+            Currently simply returns the original data. Needs to be implemented
+            """
+            return data 
+        while True:
+            try:
+                unpackagedData = unpackage(self.recvBuffer.get())
+                stateQueue.put([SM_COMMANDS.RECV_ANSIBLE, [unpackagedData]])
```

Comment:
This loop will keep unpacking the same data repeatedly, wasting CPU time. Don't do this.